### PR TITLE
fix: don't hang on unbalanced parentheses expression

### PR DESCRIPTION
**Pull request by reptee**
Hello,

Trying to create a fraction from like that `+3)/` will cause neovim to hang. This is due to while loop never stopping in case we got unbalanced expression
<https://github.com/iurimateus/luasnip-latex-snippets.nvim/blob/337c597e03e45b9ba78f016651fbfd599ebc7537/lua/luasnip-latex-snippets/math_wrA.lua#L25-L36>

Adding simple bound check is not enough, because in the edge case we would still return weird string, so instead we return empty fraction.

In short:
Before PR: `+3)/` -> *neovim hangs*
After PR: `+3)/` -> `+3)\frac{}{}`

### CODE REVIEW

**Comment by iurimateus** [comment]
Thanks! Would you mind rebasing, so it appears as one commit?

**Review by iurimateus — APPROVED** [review]
